### PR TITLE
Replace `vue-type-check` with `vue-tsc`

### PR DESCRIPTION
**Pull request by mvandenburgh**
We're currently using [`vue-type-check`](https://github.com/Yuyz0112/vue-type-check), a third party wrapper for `vetur`, to perform type-checking on our Vue templates. This PR removes that in favor of [`vue-tsc`](https://github.com/johnsoncodehk/volar/tree/master/packages/vue-tsc), which is the "official" type-checker for Vue introduced in Vue 3. It is also backward-compatible with Vue 2.7, so we can now use it as of #1258. Also, `vue-type-check` doesn't seem work with Vue 2.7, so this is a prerequisite for #1258 to be merged.

### CODE REVIEW

**Comment by mvandenburgh** [comment]
@AlmightyYakob I fixed the diff, this is ready for review now.

**Review by AlmightyYakob — APPROVED** [review]

**Comment by mvandenburgh** [comment]
I'll hold off on merging this until #1258 is merged, so that the diff of that PR isn't blown up by the changes in this one.

**Comment by mvandenburgh** [comment]
Nvm, #1258 shouldn't be merged first, since it won't pass CI until this one is merged in.

**Pull request merged**
